library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -701,11 +759,13 @@ public void emitTuples()
           }
         }
       } catch (IOException e) {
+        //An exception indicating file I/O encountered exception
+        emitEndFileControlTuple(true);
```

Comment:
If file processing failed, indicate to downstream operators that entire file is not yet processed.